controllers/appointment.js

```diff
@@ -1,45 +1,142 @@
+    return res.status(400).send({
+      message: `Content cannot be empty!`
+    })
   }
-});
-
-//GET A SINGLE APPOINTMENTS BY ATTRIBUTE
-router.get('/buildingId/:buildingId', (req,res)=> {
-    const found = appointments.some(appointment => appointment.buildingId === parseInt(req.params.buildingId));
-    if (found){
-      res.json(appointments.filter(appointment => appointment.buildingId === parseInt(req.params.buildingId)));
-    }else{
-      res.status(400).send({msg: `Appointments not found with this BuildingId: ${req.params.buildingId}`});
-    }
+
+  //Create an Appointment
+  const appointment = new Appointment({
+    id: req.body.id,
+    buildingId: req.body.buildingId,
+    boilerId: req.body.boilerId,
+    start_timestamp: req.body.start_timestamp,
+    end_timestamp: req.body.end_timestamp
   });
 
-//DELETE APPOINTMENT
-router.delete("/:id", (req, res) => {
-  const found = appointments.some(appointment => appointment.id === parseInt(req.params.id));
-  if (found) {
-    res.json({
-      msg: "Appointment Deleted",
-      appointments: appointments.filter( appointment => appointment.id !== parseInt(req.params.id))
+  //Save appointments in the database
+  appointment
+    .save(appointment)
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      res.status(500).send({
+        message: err.message || "Some error occurred while creting the appointment"
+      });
+    });
+};
+
+//Retrieve all appointments from database
+exports.findAll = (req, res) => {
+  Appointment.find({})
+    .then(data => {
+      res.send(data);
+    })
+    .catch(err => {
+      res.status(500).send({
+        message: err.message || "Some error occurred while retrieving appointments"
+      });
+    });
+};
+
+//Find a single appointment with the id
+exports.findOne = (req, res) => {
+  Appointment.findOne({
+      id: req.params.id
+    })
+    .then(data => {
+      if (!data) {
+        return res.status(404).send({
+          message: `Appointment with id: " ${req.params.id} " was not found`
+        })
+      }
+      res.send(data)
+    })
+    .catch(err => {
+      res.status(500).send({
+        message: err.message || "Some error occurred while retrieving appointments"
+      })
+    })
+};
+
+//Get single appointment buildingId
+exports.findOneBuildingId = (req, res) => {
+  Appointment.findOne({
+    buildingId: req.params.buildingId
+    })
+    .then(data => {
+      if (!data) {
+        return res.status(404).send({
+          message: `Appointment with buildingId: " ${req.params.buildingId} " was not found`
+        })
+      }
+      res.send(data)
+    })
+    .catch(err => {
+      res.status(500).send({
+        message: err.message || "Some error occurred while retrieving appointments"
+      })
+    })
+};
+
+//Update an Appointments by the id in the request
+exports.update = (req, res) => {
+  if (!req.body) {
+    return res.status(400).send({
+      message: `Data updated can not be empty!`
```

Comment:
No creo que cambie la funcionalidad pero no estoy seguro del uso de las comillas invertidas en ese mensaje.

Reply:
el uso de tiles ``se utiliza para cuando queremos agregar algo de js en el medio (una operacion o una variable) por ejemplo:
`result is ${x}`.
en tu caso deberias dejarlo con comillas simples como dijo David